.github/workflows/go.yml

```diff
@@ -0,0 +1,82 @@
+name: Go
+
+on: 
+  push:
+    branches:
+      - main
+      - release-*
+    paths:
+      - '**.go'
+      - 'go.mod'
+      - 'go.sum'
+      - '**/Dockerfile'
+  pull_request:
+    paths:
+      - '**.go'
+      - 'go.mod'
+      - 'go.sum'
+      - '**/Dockerfile'
+
+jobs:
+  prepare:
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+      hash: ${{ steps.set-hash.outputs.hash }}
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v4
+
+    - name: Populate list of images to build
+      id: set-matrix
+      run: |
+        IMAGES=($(find images -mindepth 1 -maxdepth 1 -type d -exec basename {} \;))
+        IMAGES_JSON="[$(printf '"%s",' "${IMAGES[@]}")]"
+        IMAGES_JSON="${IMAGES_JSON/,\]/]}"
+        echo "matrix={\"image\":$IMAGES_JSON}" >> $GITHUB_OUTPUT
+
+    - name: Calculate a hash for images/
+      id: set-hash
+      run: echo "hash=${{ hashFiles('images/**') }}" >> $GITHUB_OUTPUT
+
+  run_tests:
+    needs: prepare
+    runs-on: ubuntu-latest
+    strategy:
+      matrix: ${{fromJson(needs.prepare.outputs.matrix)}}
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v4
+
+    - name: Set up Go
+      uses: actions/setup-go@v4
+      with:
+        go-version-file: go.mod
+        check-latest: true
+
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v3
+
+    - name: Cache Buildx Docker layers
+      uses: actions/cache@v3
+      with:
+        path: /tmp/.buildx-cache
+        key: ${{ runner.os }}-buildx-${{ needs.prepare.outputs.hash }}
+        restore-keys: |
+          ${{ runner.os }}-buildx-
+
+    - name: Build ${{ matrix.image }}
+      uses: docker/build-push-action@v4
+      with:
+        push: false
+        tags: ${{ matrix.image }}:latest
+        context: images/${{ matrix.image }}
+        cache-from: type=local,src=/tmp/.buildx-cache
+        cache-to: type=local,dest=/tmp/.buildx-cache
+    
+    - name: Run tests
+      run: |
+        go install
+        go test -timeout 0 -v .
+        go test -timeout 0 -v ./pkg/...
+        go test -timeout 0 -v ./tests -args -image=${{ matrix.image }}
```

Comment:
Why no timeout here? From my experience, a timeout isn't a bad thing in CI environments, even if it's a rather long one.

Reply:
It's just copied from the old .travis.yml

Reply:
I see :slightly_smiling_face: